proto/delta_sharing/v1/delta_sharing.proto

```diff
@@ -174,6 +174,110 @@ message GetTableVersionResponse {
   int64 version = 1;
 }
 
+message GetTableMetadataRequest {
+  // The share name to query. It's case-insensitive.
+  string share = 1 [(buf.validate.field).string.min_len = 1];
+
+  // The schema name to query. It's case-insensitive.
+  string schema = 2 [(buf.validate.field).string.min_len = 1];
+
+  // The table name to query. It's case-insensitive.
+  string table = 3 [(buf.validate.field).string.min_len = 1];
+
+  // The response format that the client supports. The server will try to honor the client
+  // but may fallback to Parquet format.
+  repeated string response_format = 4;
+
+  // The reader features that the client supports. If the table has features that the client
+  // does not support, the server will return an error.
+  repeated string reader_features = 5;
+}
+
+message GetTableMetadataResponse {
+  uint64 version = 1;
+  oneof response {
+    ParquetWrappedResponse parquet = 2;
+    DeltaWrappedResponse delta = 3;
+  }
+}
+
+message ParquetWrappedResponse {
+  repeated ParquetWrappedAction actions = 1;
+}
+
+message ParquetWrappedAction {
+  oneof action {
+    ParquetWrappedProtocol parquet_protocol = 1;
+    ParquetWrappedMetadata parquet_metadata = 2;
+  }
+}
+
+message ParquetWrappedProtocol {
+  int32 minReaderVersion = 1;
+}
+
+message ParquetWrappedMetadata {
+  string id = 1;
+  optional string name = 2;
+  optional string description = 3;
+  optional ParquetFormat format = 4;
+  string schemaString = 5;
```

Comment:
to be consistent, we should use snake_case for the message fields. The generated serde code will handle conversion for us, as this is luckily part of the json spec for proto messages :).